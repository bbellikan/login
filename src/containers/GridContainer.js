import React from 'react';
import { connect } from 'react-redux';
import {updateColOrder, updateColWidth, deleteRow, updateCell, updateSort, deleteCol, addNew} from '../actions/grid';
import {getSortedGridData} from '../selectors';
import Grid from '../components/Grid';

const GridContainer = ({gridData, gridCols, login, updateColOrder, updateColWidth, deleteRow, updateCell, updateSort, deleteCol, addNew}) =>
    login === '' ? (null)
        :(<Grid
            gridData={gridData}
            gridCols={gridCols}
            updateColOrder = {updateColOrder}
            updateColWidth = {updateColWidth}
            deleteRow = {deleteRow}
            updateCell = {updateCell}
            updateSort = {updateSort}
            deleteCol = {deleteCol}
            addNew = {addNew}
        />);

const mapStateToProps = (state) => ({
    gridData: getSortedGridData(state),
    gridCols: state.gridCols,
    login: state.login.login
});

export default connect(
    mapStateToProps,
    {updateColOrder, updateColWidth, deleteRow, updateCell, updateSort, deleteCol, addNew}
)(GridContainer)
