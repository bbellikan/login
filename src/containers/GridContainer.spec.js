import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import GridContainer from './GridContainer';

const setup = (setupProps = {}) => {
    const gridData = [{
            type: 'POS',
            extdesc: 'ABC VONS Store 1797 SAN DIEGO CAUS',
            description: 'POS Transaction VONS Store 1797 SAN DIEGO CAUS',
            amount: '-4.43',
            differed: false,
            date: '2016-02-23'
        }];
    const gridCols = {
        "0": {
            "number": 0,
            "order": 0,
            "header": "type",
            "sort": "n",
            "deleted": false,
            "type": "STRING",
            "inputType": "text",
            "width": 150
        },
        "1": {
            "number": 1,
            "order": 1,
            "header": "extdesc",
            "sort": "n",
            "deleted": false,
            "type": "STRING",
            "inputType": "text",
            "width": 150
        },
        "2": {
            "number": 2,
            "order": 2,
            "header": "description",
            "sort": "n",
            "deleted": false,
            "type": "STRING",
            "inputType": "text",
            "width": 150
        },
        "3": {
            "number": 3,
            "order": 3,
            "header": "amount",
            "sort": "n",
            "deleted": false,
            "type": "NUMBER",
            "inputType": "text",
            "width": 150
        },
        "4": {
            "number": 4,
            "order": 4,
            "header": "differed",
            "sort": "n",
            "deleted": false,
            "type": "BOOLEAN",
            "inputType": "checkbox",
            "width": 150
        },
        "5": {
            "number": 5,
            "order": 5,
            "header": "date",
            "sort": "n",
            "deleted": false,
            "type": "DATE",
            "inputType": "date",
            "width": 150
        }
    };
    const login= setupProps.login;

    const store = configureStore()({gridData,gridCols,login});
    const wrapper = shallow(<GridContainer store={store}/>);

    return {
        store,
        wrapper
    };
};

describe('GridContainer', () => {
    test('renders without crashing', () => {
        const {wrapper} = setup({
            login: {
                login: '',
                time: new Date().getTime()
            }
        });
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.shallow().find('Grid').length).toEqual(0);

    });

    test('renders grid when loggedin without crashing', () => {
        const {wrapper} = setup({
            login: {
                login: 'login',
                time: new Date().getTime()
            }
        });
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.shallow().find('Grid').length).toEqual(1);

    });


});
