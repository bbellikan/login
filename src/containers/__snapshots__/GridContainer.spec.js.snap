// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContainer renders grid when loggedin without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(GridContainer)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "addNew": [Function],
      "deleteCol": [Function],
      "deleteRow": [Function],
      "gridCols": Object {
        "0": Object {
          "deleted": false,
          "header": "type",
          "inputType": "text",
          "number": 0,
          "order": 0,
          "sort": "n",
          "type": "STRING",
          "width": 150,
        },
        "1": Object {
          "deleted": false,
          "header": "extdesc",
          "inputType": "text",
          "number": 1,
          "order": 1,
          "sort": "n",
          "type": "STRING",
          "width": 150,
        },
        "2": Object {
          "deleted": false,
          "header": "description",
          "inputType": "text",
          "number": 2,
          "order": 2,
          "sort": "n",
          "type": "STRING",
          "width": 150,
        },
        "3": Object {
          "deleted": false,
          "header": "amount",
          "inputType": "text",
          "number": 3,
          "order": 3,
          "sort": "n",
          "type": "NUMBER",
          "width": 150,
        },
        "4": Object {
          "deleted": false,
          "header": "differed",
          "inputType": "checkbox",
          "number": 4,
          "order": 4,
          "sort": "n",
          "type": "BOOLEAN",
          "width": 150,
        },
        "5": Object {
          "deleted": false,
          "header": "date",
          "inputType": "date",
          "number": 5,
          "order": 5,
          "sort": "n",
          "type": "DATE",
          "width": 150,
        },
      },
      "gridData": Array [
        Object {
          "amount": "-4.43",
          "date": "2016-02-23",
          "description": "POS Transaction VONS Store 1797 SAN DIEGO CAUS",
          "differed": false,
          "extdesc": "ABC VONS Store 1797 SAN DIEGO CAUS",
          "type": "POS",
        },
      ],
      "login": "login",
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "updateCell": [Function],
      "updateColOrder": [Function],
      "updateColWidth": [Function],
      "updateSort": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "addNew": [Function],
        "deleteCol": [Function],
        "deleteRow": [Function],
        "gridCols": Object {
          "0": Object {
            "deleted": false,
            "header": "type",
            "inputType": "text",
            "number": 0,
            "order": 0,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          },
          "1": Object {
            "deleted": false,
            "header": "extdesc",
            "inputType": "text",
            "number": 1,
            "order": 1,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          },
          "2": Object {
            "deleted": false,
            "header": "description",
            "inputType": "text",
            "number": 2,
            "order": 2,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          },
          "3": Object {
            "deleted": false,
            "header": "amount",
            "inputType": "text",
            "number": 3,
            "order": 3,
            "sort": "n",
            "type": "NUMBER",
            "width": 150,
          },
          "4": Object {
            "deleted": false,
            "header": "differed",
            "inputType": "checkbox",
            "number": 4,
            "order": 4,
            "sort": "n",
            "type": "BOOLEAN",
            "width": 150,
          },
          "5": Object {
            "deleted": false,
            "header": "date",
            "inputType": "date",
            "number": 5,
            "order": 5,
            "sort": "n",
            "type": "DATE",
            "width": 150,
          },
        },
        "gridData": Array [
          Object {
            "amount": "-4.43",
            "date": "2016-02-23",
            "description": "POS Transaction VONS Store 1797 SAN DIEGO CAUS",
            "differed": false,
            "extdesc": "ABC VONS Store 1797 SAN DIEGO CAUS",
            "type": "POS",
          },
        ],
        "login": "login",
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "updateCell": [Function],
        "updateColOrder": [Function],
        "updateColWidth": [Function],
        "updateSort": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GridContainer renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(GridContainer)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "addNew": [Function],
      "deleteCol": [Function],
      "deleteRow": [Function],
      "gridCols": Object {
        "0": Object {
          "deleted": false,
          "header": "type",
          "inputType": "text",
          "number": 0,
          "order": 0,
          "sort": "n",
          "type": "STRING",
          "width": 150,
        },
        "1": Object {
          "deleted": false,
          "header": "extdesc",
          "inputType": "text",
          "number": 1,
          "order": 1,
          "sort": "n",
          "type": "STRING",
          "width": 150,
        },
        "2": Object {
          "deleted": false,
          "header": "description",
          "inputType": "text",
          "number": 2,
          "order": 2,
          "sort": "n",
          "type": "STRING",
          "width": 150,
        },
        "3": Object {
          "deleted": false,
          "header": "amount",
          "inputType": "text",
          "number": 3,
          "order": 3,
          "sort": "n",
          "type": "NUMBER",
          "width": 150,
        },
        "4": Object {
          "deleted": false,
          "header": "differed",
          "inputType": "checkbox",
          "number": 4,
          "order": 4,
          "sort": "n",
          "type": "BOOLEAN",
          "width": 150,
        },
        "5": Object {
          "deleted": false,
          "header": "date",
          "inputType": "date",
          "number": 5,
          "order": 5,
          "sort": "n",
          "type": "DATE",
          "width": 150,
        },
      },
      "gridData": Array [
        Object {
          "amount": "-4.43",
          "date": "2016-02-23",
          "description": "POS Transaction VONS Store 1797 SAN DIEGO CAUS",
          "differed": false,
          "extdesc": "ABC VONS Store 1797 SAN DIEGO CAUS",
          "type": "POS",
        },
      ],
      "login": "",
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "updateCell": [Function],
      "updateColOrder": [Function],
      "updateColWidth": [Function],
      "updateSort": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "addNew": [Function],
        "deleteCol": [Function],
        "deleteRow": [Function],
        "gridCols": Object {
          "0": Object {
            "deleted": false,
            "header": "type",
            "inputType": "text",
            "number": 0,
            "order": 0,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          },
          "1": Object {
            "deleted": false,
            "header": "extdesc",
            "inputType": "text",
            "number": 1,
            "order": 1,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          },
          "2": Object {
            "deleted": false,
            "header": "description",
            "inputType": "text",
            "number": 2,
            "order": 2,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          },
          "3": Object {
            "deleted": false,
            "header": "amount",
            "inputType": "text",
            "number": 3,
            "order": 3,
            "sort": "n",
            "type": "NUMBER",
            "width": 150,
          },
          "4": Object {
            "deleted": false,
            "header": "differed",
            "inputType": "checkbox",
            "number": 4,
            "order": 4,
            "sort": "n",
            "type": "BOOLEAN",
            "width": 150,
          },
          "5": Object {
            "deleted": false,
            "header": "date",
            "inputType": "date",
            "number": 5,
            "order": 5,
            "sort": "n",
            "type": "DATE",
            "width": 150,
          },
        },
        "gridData": Array [
          Object {
            "amount": "-4.43",
            "date": "2016-02-23",
            "description": "POS Transaction VONS Store 1797 SAN DIEGO CAUS",
            "differed": false,
            "extdesc": "ABC VONS Store 1797 SAN DIEGO CAUS",
            "type": "POS",
          },
        ],
        "login": "",
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "updateCell": [Function],
        "updateColOrder": [Function],
        "updateColWidth": [Function],
        "updateSort": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
